### PR TITLE
Remove lockpicking skill

### PR DESCRIPTION
**Pull request by Coolthulhu**
Lockpick skill removed. Will cause debugmsgs if mods use it.

Numbers are not tested and need tweaking. More precisely, an idea of how hard should lockpicking be.

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
Is it only dex based now?

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -2581,14 +2580,12 @@ void activity_handlers::lockpicking_finish( player_activity *act, player *p )
 
     /** @EFFECT_DEX improves chances of successfully picking door lock, reduces chances of bad outcomes */
     /** @EFFECT_MECHANICS improves chances of successfully picking door lock, reduces chances of bad outcomes */
-    /** @EFFECT_LOCKPICK greatly improves chances of successfully picking door lock, reduces chances of bad outcomes */
-    int pick_roll = std::pow( 1.5, p->get_skill_level( skill_lockpick ) ) *
+    int pick_roll = 5 *
                     ( std::pow( 1.3, p->get_skill_level( skill_mechanics ) ) +
                       it->get_quality( qual_LOCKPICK ) - it->damage() / 2000.0 ) +
                     p->dex_cur / 4.0;
```
Comment (anothersimulacrum) [inline src/activity_handlers.cpp:2586]: @Firestorm01X2 Mechanics, quality, dex, item damage

Comment (Firestorm01X2) [reply src/activity_handlers.cpp:2586]: Oh, I missed that one. Then it is great.

**Comment by DrPariah** [comment]
I like this. It's kind of a pain to just have lockpicking being a separate skill, except for roleplaying purposes. By the time you can get good at it, you'll have options like crowbars, pickaxes, and acetylene torches, most of which bypass doors _far_ more reliably than lockpicking.

**Comment by Coolthulhu** [comment]
If anyone has suggestions as to breakpoints for specific chances and time scaling, I can take them into account here.

Lockpicking design is currently broken, so I'll be leaning towards making it take more time rather than break chances.
By "broken" I mean you can open any pickable door by just spamming basic lockpick at 0 skill, it will just take a lot of keypresses. In-game time usage is pretty low. A bag of lockpicks and two bags of patience and you can open up entire prison in few hours of in-game time. You'll barely level up lockpicking from that.

There is currently no way to make certain doors easier and harder to pick. Furthermore, the list of pickable doors is hardcoded.

**Comment by Firestorm01X2** [comment]
> A bag of lockpicks and two bags of patience and you can open up entire prison in few hours of in-game time.

In this case locking up door forever can help. Similar logic as with safes. If I recall correctly original Fallout solved it in this way.

**Comment by Coolthulhu** [comment]
> In this case locking up door forever after cetain failed attempts can help.

It's hard to store data in terrain, meaning it wouldn't be "a number of attempts", but "a random chance per attempt".
And would require creating a new terrain type for each door that can get jammed.

Not worth it. Meaning we have just 3 options per character+tool pair: can't open, opens very slowly, opens fast.
There could also be "opens but destroys tool", but this only matters when the tool is costly.

**Comment by Coolthulhu** [comment]
Alright, enough time for modders to get rid of the skill from BN-ported mods.

**Comment by Night-Pryanik** [comment]
Just curious, do you really think that installing and uninstalling vehicle parts and crafting random mechanical stuff could somehow help one to be better at picking locks?

**Comment by Coolthulhu** [comment]
It's a video game, things don't need to be realistic, they need to be "fun".
Separate skill for lockpicking could make sense in a game with multiple grades of doors, stealth sections, skill level-ups allocated from pool instead of learn-by-doing.
Having a separate skill just for lockpicking in an open world game with electric jackhammers, powerful explosives, and low penalties for making noise, just didn't add anything to the game.

Mechanics is the most relevant non-redundant skill here. I left bonus from mechanics in because I just wanted to remove the lockpicking skill, not rebalance the whole lockpicking.
Now, it could use this rebalancing, because the ability to open any pickable door by just spamming the weakest grade lockpick is bad, and this rebalancing could involve removing the mechanics bonus, but it's just not in scope of this PR.

**Comment by Night-Pryanik** [comment]
I can understand the "Fun uber alles" principle of your fork, but honestly I see no particular fun in removing working skills (albeit with niche use). I can't understand how removing _anything_ can increase fun at all.
Moreover, how could one add new functionality to the lockpicking (e.g. several grades of locked doors aforementioned by you) or balance penalties for making noise, if lockpicking is gone for good after only 5 months since implementation?

**Comment by Coolthulhu** [comment]
> I can't understand how removing anything can increase fun at all.

Having a skill that covers only very specific things (for example, computers, banter, speaking), and thus is trained and used in a very limited set of situations, leads to "grind or don't use at all" choice. That's anti-fun.

Having the same very specific situations be covered by already useful skills/stats/traits can lead to interesting choices.
Say, if perception governed lockpicking and stealth, high perception characters would be natural night raiders, being safer in the darkness but subpar in the daylight.
A great example is cooking vs chemistry. BN only has cooking, because chemistry lacks uses at low levels, while cooking lacks uses at high level, so keeping them as one skill is the perfect solution. They also use some common ingredients, which is a bonus.

> Moreover, how could one add new functionality to the lockpicking (e.g. several grades of locked doors aforementioned by you) or balance penalties for making noise, if lockpicking is gone for good after only 5 months since implementation?

It might be reintroduced later if the niche for such a skill is carved out.
Until then, grades of doors could just require different amounts of dexterity/perception, better tools, prohibitive amount of time if attempted with stats/tools too bad, strong light (no stealth). Also, a very important thing would be more doors/walls that can't be jackhammered and torched.
Penalties for making noise can be considered without thinking about lockpicking specifically.
But then, it might work perfectly fine without a skill for itself. Parkour expert is not a skill, even though it could be.

**Comment by Night-Pryanik** [comment]
>Having a skill that covers only very specific things (for example, computers, banter, speaking), and thus is trained and used in a very limited set of situations, leads to "grind or don't use at all" choice. That's anti-fun.

Lockpicking is a bad example here. You don't need to grind lockpicking (or mechanics, in your fork) to be able to use it. You can successfully pick any lock even with the shittiest lockpick if you're lucky and/or patient enough. It's the issue with the lockpicking mechanics itself, not the skill that governs it. And thus by removing the lockpicking skill you don't make the lockpicking mechanics itself less "anti-fun".

On the other hand, using your reasoning, we should remove all the combat skills altogether. They all are perfect examples of "grind or don't use at all", as usually it's a very bad idea to go into melee or fire some ranged weapon at low levels.

>cooking, because chemistry lacks uses at low levels, while cooking lacks uses at high level, so keeping them as one skill is the perfect solution.

Using this logic, you can substitute cooking with any other skill that doesn't have uses at high levels (even driving as an absurd example).

**Comment by Coolthulhu** [comment]
> And thus by removing the lockpicking skill you don't make the lockpicking mechanics itself less "anti-fun".

I do: by making lockpicking easier (substituting 4 for lockpicking skill), I made it less repetitive. The lockpicking mechanics are still horrid, but now the player can get over them faster, with less keypresses.
Also, it makes it easier to expand lockpicking in the future, as there is no risk of running into "you have to grind lockpicking to lvl x before proceeding" if lockpicking difficulty is increased to make it meaningful. Lockpicking practice mechanics were also a pain, I'd much rather just have a flat-out "no, this character won't ever pick this lock" instead of grind here.

> They all are perfect examples of "grind or don't use at all", as usually it's a very bad idea to go into melee or fire some ranged weapon at low levels.

Melee skills level up naturally, without the need for explicit grind.
They are used everywhere, the cost of failure is non-negligible but not "infinite", there is no hard threshold for them. At the same time, you can't just retry using them over and over until you succeed, because the cost of retrying is high.
Ranged weapons are worse in this regard, but BN doesn't have the problem of archery being useless at low levels. They are usable without explicit grind, but also benefit less from grind.

Meanwhile lockpicking was a "perfectly useless" skill: it only saves you minor amount of time and cheap items.
Saving player from frustration is not a balancing factor, a well designed game will always try to minimize frustration and never use it for balance.

> Using this logic, you can substitute cooking with any other skill that doesn't have uses at high levels (even driving as an absurd example).

Cooking and chemistry share a lot of traits and even tools and low/high lvl thing is basically their only major difference.

**Comment by Night-Pryanik** [comment]
> I made it less repetitive.

Ok, I think I got it. It seems that "fun" in your understanding means simply "faster to play". If so, your definition of "fun" differs from mine.
While I'm not neglecting that fast-paced gameplay could be fun, I'm definitely neglecting that fast-paced gameplay is _the_ fun, and that literally every other aspect of gameplay could be sacrificed in order not to slow it down.

> Melee skills level up naturally, without the need for explicit grind.

The same for lockpicking skill.

≥Meanwhile lockpicking was a "perfectly useless" skill: it only saves you minor amount of time and cheap items.

This is more about lockpicking mechanics in general rather than about having a separate skill for it. If there are so many ways to bypass locked doors, and most people don't even bother with lockpicking, we could as well remove the whole lockpicking mechanics, and no one even notice that.

**Comment by Coolthulhu** [comment]
> Ok, I think I got it. It seems that "fun" in your understanding means simply "faster to play". If so, your definition of "fun" differs from mine.

No, it's more about only having to do the actions that matter.
Repeated trivial actions aren't fun. Lockpicking involves spamming e->direction->direction->i over and over again.

> The same for lockpicking skill.

Read the rest of the paragraph I wrote on why does melee learn naturally and why lockpicking doesn't.

> If there are so many ways to bypass locked doors, and most people don't even bother with lockpicking, we could as well remove the whole lockpicking mechanics, and no one even notice that.

Currently lockpicking is important for prison starts. So I decided to make it less annoying rather than removing it completely.

> This is more about lockpicking mechanics in general rather than about having a separate skill for it.

Separate skill contributes to the problem by making lockpicking more tedious to use when it's actually useful.

**Comment by neuks** [comment]
I think the BN version is not only more fun, but also more reasonably sane than the original one, I've switched to this version to play already.

**Comment by Firestorm01X2** [comment]
@Coolthulhu 
Skill still listed here:
https://github.com/cataclysmbnteam/Cataclysm-BN/blob/upload/data/json/skills.json#L282

Also this:
https://github.com/cataclysmbnteam/Cataclysm-BN/issues/78

**Comment by Night-Pryanik** [comment]
> Lockpicking involves spamming e->direction->direction->i over and over again.

Once again, it's the issue of lockpicking mechanics, not the separate skill used for it. If lockpicking is unfun, then it should be reworked or removed - it doesn't matter which skill governs it.

>Separate skill contributes to the problem by making lockpicking more tedious to use when it's actually useful.

I simply don't get how grinding mechanics to more effectively pick locks is less tedious and more fun than grinding lockpicks to do the same.

Feels like we're heading in circles here. I heard your reasonings, you heard mine. Seems like we have totally different views on game design, gameplay and definitions of fun in the game.

**Comment by Coolthulhu** [comment]
> I simply don't get how grinding mechanics to more effectively pick locks is less tedious and more fun than grinding lockpicks to do the same.

Lockpicking doesn't grind mechanics. I didn't add the bonus from mechanics, it was already there.

**Comment by Firestorm01X2** [comment]
@Coolthulhu 
Hotfix ready:
https://github.com/cataclysmbnteam/Cataclysm-BN/pull/79

**Comment by Night-Pryanik** [comment]
I meant that you have to grind mechanics beforehand to not to suck at lockpicking.

**Comment by Coolthulhu** [comment]
> I meant that you have to grind mechanics beforehand to not to suck at lockpicking.

Mechanics is useful in other ways and grinding it involves gathering resources and producing useful things (cart for hauling).
It's not in the same tier of evil as grinding a skill by safe, tedious, repeated applications of it.

If I get to rebalancing lockpicking to make it sensible, I'll probably remove that bonus from mechanics. But for now, the goal was removing the biggest offender.

**Pull request merged**